src/ClientBuilder.php

```diff
@@ -175,19 +175,19 @@ private function createTransportInstance(): TransportInterface
      */
     private function createDefaultTransportFactory(): DefaultTransportFactory
     {
-        $streamFactory = Psr17FactoryDiscovery::findStreamFactory();
+        $psr17Factory = new WellKnownPsr17Factory();
         $httpClientFactory = new HttpClientFactory(
-            Psr17FactoryDiscovery::findUriFactory(),
-            Psr17FactoryDiscovery::findResponseFactory(),
-            $streamFactory,
+            null,
+            null,
```

Comment:
The first two arguments of `HttpClientFactory` are never used. We can thus make them nullable and skip injecting and factory there.